### PR TITLE
ChatAPI functionality

### PR DESCRIPTION
**Pull request by sotoz**
Added actions for Chat API.
- Chat Channels Create/Delete/Update/List all/View one
- Chat Contacts Create/Delete/List all/View one
- Chat Messages Create/List all/View one
- Chat Platforms List all

Created examples for every needed action.

### CODE REVIEW

**Comment by samwierema** [comment]
My first instinct is that I would separate all Chat API-related objects and resources into their own sub-namespace. So where it's now `MessageBird\Objects\ChatChannel` I would feel we should make clear it's a separate API (and can thus behave differently) by putting it in it's sub-namespace, e.g.: `MessageBird\Objects\Chat\Channel`.

**Comment by bviolier** [comment]
@samwierema Is that also the case for the library? I mean, does the user care that it can behave differently deep down in the code, if on the surface it works the same for him?

**Comment by bviolier** [comment]
I agree with @samwierema

**Comment by sotoz** [comment]
Should we do it only for the Chat related objects?

**Comment by samwierema** [comment]
@sotoz yes, everything else should remain in place.

File: examples/chatchannels-create.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatchannels-create.php]: Dummy access key contains tab character.

File: examples/chatchannels-delete.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatchannels-delete.php]: Dummy access key contains tab character.

File: examples/chatchannels-list.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatchannels-list.php]: Dummy access key contains tab character.

File: examples/chatchannels-view.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatchannels-view.php]: Dummy access key contains tab character.

File: examples/chatcontacts-list.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatcontacts-list.php]: Dummy access key contains tab character.

File: examples/chatcontacts-view.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatcontacts-view.php]: Dummy access key contains tab character.

File: examples/chatmessages-create.php [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatmessages-create.php]: Dummy access key contains tab character.

File: examples/chatmessages-view.php [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatmessages-view.php]: Dummy access key contains tab character.

File: examples/chatplatforms-list.php [hunk 9]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY	'); // Set your own API access key here.
```
Comment (dstotijn) [inline examples/chatplatforms-list.php]: Dummy access key contains tab character.

**Comment by dstotijn** [comment]
Please also check for double newlines (maybe run phpcs with PSR2)?

File: src/MessageBird/Client.php [hunk 10]
```diff
@@ -11,6 +11,7 @@ class Client
 {
 
     const ENDPOINT = 'https://rest.messagebird.com';
+    const CHATAPI_ENDPOINT = 'https://chat.messagebird.com/1';
 
     const CLIENT_VERSION = '1.4.1';
```
Comment (ged15) [inline src/MessageBird/Client.php]: should this be updated during release somehow?

File: src/MessageBird/Objects/Chat/Channel.php [hunk 11]
```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace MessageBird\Objects\Chat;
+
+use MessageBird\Objects\Base;
+
+/**
```
Comment (ged15) [inline src/MessageBird/Objects/Chat/Channel.php]: we should avoid meaningless docblocks

File: src/MessageBird/Objects/Chat/Channel.php [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+     * @var array
+     */
+    protected $_links = array();
+
+    /**
+     * @param $object
+     *
+     * @return $this|void
+     */
+    public function loadFromArray ($object)
```
Comment (ged15) [inline src/MessageBird/Objects/Chat/Channel.php]: shouldn't the parameter be named `$array` and typehinted to `array`?

File: src/MessageBird/Resources/Base.php [hunk 13]
```diff
@@ -171,5 +171,28 @@ public function processRequest($body)
         throw new Exceptions\RequestException($ResponseError->getErrorString());
     }
 
+    /**
+     * @param $object
+     * @param $id
+     * @return $this ->Object
+     * @internal param array $parameters
```
Comment (ged15) [inline src/MessageBird/Resources/Base.php]: unneeded

File: examples/chatchannels-create.php [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+require_once(__DIR__ . '/../autoload.php');
+
+$MessageBird = new \MessageBird\Client('YOUR_ACCESS_KEY'); // Set your own API access key here.
+
+$ChatChannel = new \MessageBird\Objects\Chat\Channel();
+
+//Example for telegram channel
+
+$ChatChannel->name = 'Test Channel Telegram';
+$ChatChannel->platform_id = 'e82d332c5649a5f911e569n69040697';
```
Comment (ged15) [inline examples/chatchannels-create.php]: this should be `platformId`

**Comment by ged15** [comment]
no example to create contacts?

**Comment by sotoz** [comment]
`platformId` will be corrected. 
Example for creating contacts not needed. Contacts are being created automatically when the first  message arrives.

**Pull request merged**